mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp

```diff
@@ -3269,15 +3269,20 @@ DiagnosedSilenceableFailure transform::FlattenElementwiseLinalgOp::applyToOne(
     transform::ApplyToEachResultList &results,
     transform::TransformState &state) {
   rewriter.setInsertionPoint(target);
-  if (target.getNumLoops() <= 1)
+  if (!isElementwise(target)) {
+    failed(rewriter.notifyMatchFailure(
+        target, "only elementwise flattening is supported"));
+    return emitDefaultSilenceableFailure(target);
```

Comment:
Instead of sending the error message to the non-existent rewrite driver, this should rather emit it as a silenceable failure! The default failure messages produced by `emitDefaultSilenceableFailure` are extremely unhelpful and must only be used as last resort.

I also haven't seen the usage of `failed` as a way to suppress the compiler warning about `LogicalResult` being unused. The common idiom is to use the C-style cast to void.

Reply:
Sorry, I'm new to the transform framework and still learning how things work. I was just following another implementation in this same file for the message. I'll submit another PR to fix.

About tagging you.  I made an RFC for this per your suggestion with a mention to the original PR and Mahesh was the only one that took any action so I just went with him again for this.  I'll be sure to include you in the future.

Reply:
No big deal, the point of review is also for contributors to learn things.

Reply:
appreciate it. thanks for teaching me. :)